pyro/infer/rws.py

```diff
@@ -0,0 +1,227 @@
+import math
+import warnings
+
+import torch
+
+import pyro
+import pyro.poutine as poutine
+from pyro.infer.elbo import ELBO
+from pyro.infer.enum import get_importance_trace
+from pyro.infer.util import is_validation_enabled
+from pyro.poutine.util import prune_subsample_sites
+from pyro.util import check_if_enumerated, check_model_guide_match, warn_if_nan
+
+
+class ReweightedWakeSleep(ELBO):
+    r"""
+    An implementation of Reweighted Wake Sleep following reference [1].
```

Comment:
Could you also add a comment explaining that the guide needs to take an optional kwarg `observations` which is a dict?

Reply:
Done in `7532af20`.